### PR TITLE
Bug 1905348: Gherkin Script Scenarios for epic ODC-4658

### PR DESCRIPTION
**Pull request by gajanan-more**
Fixes: https://issues.redhat.com/browse/ODC-5216

Problem: Gherkin Scripts for scenarios for [ODC-4658](https://issues.redhat.com/browse/ODC-4658)

Solution: These are the gherkin scripts designed covering all the scenarios related to epics [ODC-4658](https://issues.redhat.com/browse/ODC-4658)

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/operators.feature [hunk 1]
```diff
@@ -126,3 +126,24 @@ Scenario: Install Web Terminal operator from Operator Hub page
...
+@regression, @smoke
+Scenario: Create Integration Platform CR
+   Given user has installed OpenShift Serverless Operator
+   And user has installed Red Hat Integration - Camel Operator
+   And user is on Installed Operator page
+   When user clicks on Integration Platform link
+   And user clicks on Create IntegrationPlatform button
+   And user clicks on Create button
+   Then user will be redirected to Integration Platform tab with header "IntegrationPlatforms"
+   And user will see Integration Platform created with name example
```
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/operators.feature:149]: Added

File: frontend/packages/dev-console/integration-tests/features/operators.feature [hunk 2]
```diff
@@ -126,3 +126,24 @@ Scenario: Install Web Terminal operator from Operator Hub page
    And user installs the Web Terminal operator with default values
    Then user will see a modal with title "Web Terminal"
    And user will see a View Operator button
+
+
+@regression, @smoke
+Scenario: Install Red Hat Integration - Camel Operator
+   Given user has installed OpenShift Serverless Operator
+   And user is at Operator Hub page with the header name "OperatorHub"
+   When user search and installs the Red Hat Integration - Camel Operator with default values
```
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/operators.feature:135]: Updated

File: frontend/packages/dev-console/integration-tests/features/operators.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/operators.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/operators.feature (hunk 2 above)
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/operators.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature [hunk 3]
```diff
@@ -0,0 +1,24 @@
+Feature: Creation of Kamelets via Event Source Creation
+    As a user, I should be able to install and use more camel connectors
+
+
+    Background:
+        Given user has installed OpenShift Serverless Operator
+        And user has created Knative Serving and Knative Eventing CR
+        And user has installed Red Hat Integration - Camel Operator
+        And user has created Integration Platform CR
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature:9]: shall we mention namespace here global installation is not there , will see kamelets only in ns we added CR for IP

File: frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+
+    @regression
+    Scenario: Kamelets on Search page
+        Given user is at Developer Perspective
+        When user goes to Search page
+        And user checks Kamlet checkbox
+        Then user will see aws-kinesis-source, aws-sqs-source, jira-source, salesforce-source, slack-source, telegram-source kamelets installed
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature:17]: we may not need this , if kamelets is there should show up under eventSources which you already covered below

File: frontend/packages/dev-console/integration-tests/features/operators.feature [hunk 5]
```diff
@@ -126,3 +126,24 @@ Scenario: Install Web Terminal operator from Operator Hub page
...
    Then user will see a modal with title "Web Terminal"
    And user will see a View Operator button
+
+
+@regression, @smoke
+Scenario: Install Red Hat Integration - Camel Operator
+   Given user has installed OpenShift Serverless Operator
+   And user is at Operator Hub page with the header name "OperatorHub"
+   When user searches and installs the Red Hat Integration - Camel Operator with default values
+   Then user will see a modal with title "knative Apache Camel Operator"
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/operators.feature:136]: should this be `Red Hat Integration - Camel K`?

File: frontend/packages/dev-console/integration-tests/features/operators.feature [hunk 6]
```diff
@@ -126,3 +126,24 @@ Scenario: Install Web Terminal operator from Operator Hub page
...
+   And user will see a View Operator button
+
+
+@regression, @smoke
+Scenario: Create Integration Platform CR
+   Given user has installed OpenShift Serverless Operator
+   And user has installed Red Hat Integration - Camel Operator
+   And user is on Installed Operator page
+   When user clicks on Integration Platform link
+   And user clicks on Create IntegrationPlatform button
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/operators.feature:146]: this had relationship with namespace it's created in

File: frontend/packages/dev-console/integration-tests/features/operators.feature (hunk 5 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/operators.feature:136]: Yeah!!!. Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature (hunk 4 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature:17]: Removed the scenario.

File: frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature (hunk 3 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature:9]: Added the namespace in the background which will run before every scenario.

File: frontend/packages/dev-console/integration-tests/features/operators.feature (hunk 6 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/operators.feature:146]: Added the namespace in the background which will run before every scenario.

File: frontend/packages/dev-console/integration-tests/features/operators.feature (hunk 5 above)
Comment (invincibleJai) [reply frontend/packages/dev-console/integration-tests/features/operators.feature:136]: `Then user will see a modal with title "knative Apache Camel Operator"` i don't see this updated

File: frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature (hunk 3 above)
Comment (invincibleJai) [reply frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature:9]: sorry din't got what you mean by background here, can you point to that ??

File: frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature [hunk 7]
```diff
@@ -0,0 +1,17 @@
+Feature: Creation of Kamelets via Event Source Creation
+    As a user, I should be able to install and use more camel connectors
+
+
+    Background:
+        Given user has installed OpenShift Serverless Operator
+        And user has created Knative Serving and Knative Eventing CR
+        And user has installed Red Hat Integration - Camel K Operator
+        And user has created Integration Platform CR
+        And user has selected "aut-test-kamelets" namespace
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature:10]:
```suggestion
        And user has selected "aut-test-kamelets" namespace
        And user has created Integration Platform CR
```

should it be like this i.e ns should be selected before creating IP?

File: frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature (hunk 3 above)
Comment (invincibleJai) [reply frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature:9]: ignore i see this here https://github.com/openshift/console/pull/7441#discussion_r539847326

File: frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature (hunk 7 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature:10]: It's correct Jai. For the statetment `user has created Integration Platform CR`, scenario from operators.feature will run in the background.

Comment (invincibleJai) [reply frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature:10]: which ns user is in when `user has created Integration Platform CR` ??

Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature:10]: `"aut-test-kamelets"`

Comment (invincibleJai) [reply frontend/packages/dev-console/integration-tests/features/Knative/eventing-kamletes.feature:10]: so as you described this happens here https://github.com/openshift/console/pull/7441/files#diff-072029627bf4da5d30ef8de570b1ac4f646eb59ffd5f2a2f2516db68135c256aR141 so adding it here again may not be needed

```
 And user has selected "aut-test-kamelets" namespace
```

**Pull request merged**